keps/sig-cli/3655-kubectl-translation-improvement/README.md

```diff
@@ -0,0 +1,1225 @@
+
+| Locale | Reviewers | Approvers |
+|--------|-----------|-----------|
+| de_DE  | @iamNoah1 | @iamNoah1 |
+| fr_FR  |           |           |
+| it_IT  |           |           |
+| ja_JP  |           |           |
+| ko_KR  | @seokho-son | @seokho-son |
+| pt_BR  |           |           |
+| zh_CN  |           |           |
+| zh_TW  |           |           |
+
+<<[UNRESOLVED TODO]>>
+
+TODO: 
+* Identify reviewers and approves for each locale and populate the able above
+
+<</[UNRESOLVED]>>
+
+### 3. DETAILS: Improve documentation.
+
+There is a [`README.md`](https://github.com/kubernetes/kubernetes/blob/eb75b34394964fbc325ca08a5d0b126c5ea16b6f/staging/src/k8s.io/kubectl/pkg/util/i18n/translations/README.md) located within kubectl's i18 directory, which is helpful, but the process is not simple for a new or occasional contributor.
+
+This document will be updated to remove any out-of-date information and add any new information.
+
+We will also create a separate page on [k8s.dev](https://k8s.dev) and link to it from [Non-code Contributions](https://www.k8s.dev/docs/guide/non-code-contributions/), so that people who are interested in becoming a translator for kubectl can discover how to start.
+
+### 4. DETAILS: Ensure all strings in code support translations.
+
+Create an issue to track the work needed to wrap strings in `i18n.T()`.
+
+Fix known issues affecting translation completeness:
+* https://github.com/kubernetes/kubectl/issues/1326
+* https://github.com/kubernetes/kubectl/issues/1327
+
+<<[UNRESOLVED TODO]>>
+
+TODO:
+* There was already an issue for this, but it is closed now: https://github.com/kubernetes/kubectl/issues/910
+* There is a [script called `extract.py` in the translations directory](https://github.com/kubernetes/kubernetes/blob/5ece28b77a284b24b674278378630373196789ac/staging/src/k8s.io/kubectl/pkg/util/i18n/translations/extract.py), but it doesn't seem to work anymore.
+  * Should we try to get this back in working order?
+  * How can it be used? Can we make a `hack/extract-translations.sh` or something like that to standardize its usage?
+
+<</[UNRESOLVED]>>
+
+### 5. DETAILS: Establish metrics to track the state of translation completeness.
+
+In order to understand the current state of translations and track progress or regression over time, we need to establish a way to measure how complete the translations are for each locale.
+
+Prototype utility to analyze `.po`/`.pot` files to determine translation coverage:  
+https://github.com/brianpursley/gettext-report
+
+<<[UNRESOLVED TODO]>>
+
+TODO:
+* How can the process be run automatically? 
+* Should the data be stored over time or just be presented as a snapshot in time? 
+  * If stored over time, where can this be stored? 
+  * Are there other things in the k8s codebase that generate metrics that can be used as an example?
+* Can we use periodic tests that run and fail if there are any incomplete translations?
+  * Is the threshold 100% or some other specific %? 
```

Comment:
If we go with a supported and unsupported lists, I'd start initially with 50%, we can adjust (and that should be called out explicitly here) over time, as the translations improve.